MekHQ/src/mekhq/util/dom/DomProcessor.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * MekHQ - Copyright (C) 2018 - The MekHQ Team
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software
+ * Foundation; either version 2 of the License, or (at your option) any later
+ * version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details.
+ */
+package mekhq.util.dom;
+
+import static java.util.stream.Collectors.toSet;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+/**
+ * Convenience class to process the children (elements and attributes) of an xml element.
+ * <p>
+ * Unless otherwise noted, all methods respect document-order.
+ */
+public class DomProcessor {
```

Comment:
IMHO this makes parsing sooooo much easier - see for example `ForceStub.generateInstanceFromXML()`.

Let me know what you think of this class: I can remove it and retrofit the few `generateInstanceFromXML` where I've used it or go on and (slowly) spread its use all over the place.